File: e2e/elixir/test/batch_test.exs
# E2e tests for category: batch
defmodule E2e.BatchTest do
  use ExUnit.Case, async: true

  describe "scrape_batch_basic" do
    test "Batch scrape of multiple URLs all succeeding" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(result.batch.completed_count) == 3
      assert String.trim(result.batch.failed_count) == 0
      assert String.trim(result.batch.total_count) == 3
    end
  end

  describe "scrape_batch_partial_failure" do
    test "Batch scrape with one URL failing returns partial results" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(result.batch.completed_count) == 2
      assert String.trim(result.batch.failed_count) == 1
      assert String.trim(result.batch.total_count) == 3
    end
  end

  describe "scrape_batch_progress" do
    test "Batch scrape results include specific URL" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(result.batch.total_count) == 2
      assert String.contains?(result.batch.results, "/target")
    end
  end
end

File: e2e/go/batch_test.go
// E2e tests for category: batch
package e2e_test

import (
	"strings"
	"testing"

	pkg "github.com/kreuzberg-dev/kreuzcrawl"
)

func Test_ScrapeBatchBasic(t *testing.T) {
	// Batch scrape of multiple URLs all succeeding
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'batch.completed_count' not available on result type
	// skipped: field 'batch.failed_count' not available on result type
	// skipped: field 'batch.total_count' not available on result type
}

func Test_ScrapeBatchPartialFailure(t *testing.T) {
	// Batch scrape with one URL failing returns partial results
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'batch.completed_count' not available on result type
	// skipped: field 'batch.failed_count' not available on result type
	// skipped: field 'batch.total_count' not available on result type
}

func Test_ScrapeBatchProgress(t *testing.T) {
	// Batch scrape results include specific URL
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'batch.total_count' not available on result type
	// skipped: field 'batch.results' not available on result type
}
